roachtest: deflake `failover/chaos` deadlock failures

erikgrinaker opened the pull request:
The deadlock failure mode used `SHOW RANGES` to fetch range and lease information from the cluster. However, if this was run concurrently with other failures in chaos tests it could hang and time out, since it needed to read lease information from each individual range.

This patch restructures the deadlock failure mode to fetch range/lease information from the cluster before running failures, and skipping any replicas that have since moved.

Resolves #104045.
Resolves #104086.
Resolves #104140.
Resolves #104144.

Epic: none
Release note: None

erikgrinaker committed:
roachtest: resume dmsetup disk staller during cleanup

If the test failed while the disk was stalled, the cleanup step wouldn't
be able to proceed. This patch explicitly resumes the disk before
cleanup.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/104138)
<!-- Reviewable:end -->

erikgrinaker committed:
roachtest: deflake `failover/chaos` deadlock failures

The deadlock failure mode used `SHOW RANGES` to fetch range and lease
information from the cluster. However, if this was run concurrently with
other failures in chaos tests it could hang and time out, since it
needed to read lease information from each individual range.

This patch restructures the deadlock failure mode to fetch range/lease
information from the cluster before running failures, and skipping any
replicas that have since moved.

Epic: none
Release note: None

tbg commented on pkg/cmd/roachtest/tests/failover.go:
```diff
@@ -253,7 +255,7 @@ func runFailoverChaos(
 					}
 				}
 				if d, ok := failer.(*deadlockFailer); ok { // randomize deadlockFailer
-					d.numReplicas = 1 + rng.Intn(3)
```
Random fix?

erikgrinaker commented:
TFTR!

bors r+
